### ISSUE TITLE
2FA gives blank screen when logging on

### ISSUE THREAD

**Issue by NickJH** [eglooca/app-two-factor-auth#4]
Once it is installed, any time you log in you end up with a blank page at https://clearos_IP:port/app/base/session/login/. [^] You are logged in but have to edit the url to get to a sensible webconfig screen. The 2FA issue happens whether or not you have it enabled for any account.

I have reproduced this on a couple of installations, but I understand the app may be working where it has been installed for a while.

Reported at Clearcenter as https://tracker.clearos.com/view.php?id=22781

**Comment by NickJH** [comment]
Hmm. Installed on another machine. It may be the accounts manager issue as a machine I was testing on I was also testing AD DC logins and it may have had issues jointing the domain. I cannot get back to the test machine now.
The root account should work without the accounts manager, but obviously a proper user requires the accounts manager.

**Comment by bchambers** [comment]
MFA app is an extension, not a plugin. As such, it only supports OpenLDAP engine, not the AD connector.

Marketplace app rules need updating to reflect this:

/usr/clearos/apps/marketplace/deploy/cart_rules.php

It should not bomb out if installed, though.  I will add a check.

**Comment by NickJH** [comment]
Hmm, shouldn't it still work for root even with AD connector as this is outside LDAP/AD set up.

**Comment by bchambers** [comment]
Yes, it should, but it wasn't scoped/designed with the AD engine at all.  There's been zero testing and you've found a bug.   Likely some function such as get_user_2fa_list that uses the user factory is borked.

**Comment by NickJH** [comment]
Trying to re-interpret my notes of November, I think I may mean root logins don't work with or without users enabled for 2FA. I probably didn't test a user login at all, just root login. It is hard to retrace my steps as it is in a non-functioning VM.
